src/components/settings/SignerAccountMFA/index.tsx

```diff
@@ -1,31 +1,79 @@
+import { Box, Button, Checkbox, FormControlLabel, TextField, Typography } from '@mui/material'
 import { COREKIT_STATUS } from '@web3auth/mpc-core-kit'
 import { getPubKeyPoint } from '@tkey-mpc/common-types'
-import useMFASettings from './useMFASettings'
 import { BN } from 'bn.js'
-import { useState } from 'react'
+import { useEffect, useMemo, useState } from 'react'
+import { useSecurityQuestions } from '@/hooks/wallets/mpc/recovery/useSecurityQuestions'
+import useMFASettings from './useMFASettings'
+import { useForm } from 'react-hook-form'
+import { useDeviceShare } from '@/hooks/wallets/mpc/recovery/useDeviceShare'
+
+type SignerAccountFormData = {
+  oldPassword: string | undefined
+  newPassword: string
+  confirmPassword: string
+  storeDeviceShare: boolean
+}
 
 const SignerAccountMFA = () => {
   const mpcCoreKit = useMPC()
-  const mfaSettings = useMFASettings()
+  const mfaSettings = useMFASettings(mpcCoreKit)
+  const securityQuestions = useSecurityQuestions(mpcCoreKit)
+  const deviceShareModule = useDeviceShare(mpcCoreKit)
+
+  const formMethods = useForm<SignerAccountFormData>({
+    mode: 'all',
+  })
+
+  const { register, formState, watch, setValue, handleSubmit } = formMethods
 
   const [enablingMFA, setEnablingMFA] = useState(false)
 
+  const isPasswordSet = useMemo(() => securityQuestions.isEnabled(), [securityQuestions])
+
+  console.log(mpcCoreKit)
+
+  useEffect(() => {
+    deviceShareModule.isEnabled().then((value) => setValue('storeDeviceShare', value))
+  }, [deviceShareModule, setValue])
+
   const enableMFA = async () => {
     if (!mpcCoreKit) {
       return
     }
 
     setEnablingMFA(true)
     try {
-      // First enable MFA in mpcCoreKit
-      const recoveryFactor = await mpcCoreKit.enableMFA({})
+      const { newPassword, oldPassword, storeDeviceShare } = formMethods.getValues()
+      // 1. setup device factor with password recovery
+      await securityQuestions.upsertPassword(newPassword, oldPassword)
+      const securityQuestionFactor = await securityQuestions.recoverWithPassword(newPassword)
+      if (!securityQuestionFactor) {
+        throw Error('Could not recover using the new password recovery')
+      }
 
-      // Then remove the recovery factor the mpcCoreKit creates
-      const recoverKey = new BN(recoveryFactor, 'hex')
-      const recoverPubKey = getPubKeyPoint(recoverKey)
-      await mpcCoreKit.deleteFactor(recoverPubKey, recoverKey)
+      if (!mfaSettings?.mfaEnabled) {
+        // 2. enable MFA in mpcCoreKit
+        const recoveryFactor = await mpcCoreKit.enableMFA({})
+
+        // 3. remove the recovery factor the mpcCoreKit creates
+        const recoverKey = new BN(recoveryFactor, 'hex')
+        const recoverPubKey = getPubKeyPoint(recoverKey)
+        await mpcCoreKit.deleteFactor(recoverPubKey, recoverKey)
+      }
+
+      const hasDeviceShare = await deviceShareModule.isEnabled()
+      if (hasDeviceShare !== storeDeviceShare) {
+        if (storeDeviceShare) {
```

Comment:
This nested ondition is a bit hard to understand imo. Is it possible to have 2 separate conditions here? i.e. `if (!hasDeviceShare && storeDeviceShare)` and `if (hasDeviceShare && !storeDeviceShare)`